src/OmnichannelChatSDK.ts

```diff
@@ -779,38 +780,60 @@ class OmnichannelChatSDK {
         }
     }
 
-    public async endChat(): Promise<void> {
-        this.scenarioMarker.startScenario(TelemetryEvent.EndChat, {
+    private async closeChat(endChatOptionalParams: EndChatOptionalParams): Promise<void> {
+
+        const cleanupMetadata = {
             RequestId: this.requestId,
-            ChatId: this.chatToken.chatId as string
-        });
+            ChatId: this.chatToken.chatId as string,
+        };
 
-        const sessionCloseOptionalParams: ISessionCloseOptionalParams = {};
+        // in case a session was ended by agent or disconnected, there is no need to close the session
+        this.scenarioMarker.startScenario(TelemetryEvent.CloseChatSession, cleanupMetadata);
 
-        if (this.isPersistentChat && !this.chatSDKConfig.persistentChat?.disable) {
-            const isReconnectChat = this.reconnectId !== null ? true : false;
+        if (!endChatOptionalParams?.isSessionEnded) {
```

Comment:
should we add isSessionEnded to telemetry so that it is easier to distinguish between customer and agent